src/monodroid/jni/Android.mk

```diff
@@ -3,9 +3,9 @@ LOCAL_PATH := $(call my-dir)
 include $(CLEAR_VARS)
 
 ifeq ($(CONFIGURATION),Debug)
-COMMON_CFLAGS ?= -ggdb3 -O0 -fno-omit-frame-pointer
+COMMON_CFLAGS ?= -ggdb3 -O0 -fno-omit-frame-pointer -DDEBUG=1
 else
-COMMON_CFLAGS ?= -g -O2 -DRELEASE=1
```

Comment:
because this way Application.mk is different for libmono-android.debug.*so and libmono-android.release.*so

before we ended with libmono-android.debug.*so which was build with RELEASE=1 and thus was unable to load libmonosgen from shared runtime for example